### ISSUE TITLE
:bulb: [Feature] GeoServer security vulnerabilities

### ISSUE THREAD

**Issue by fmigneault** [bird-house/birdhouse-deploy#320]
## Description

I ran a security vulnerabilities scan against the new
`pavics/geoserver:2.22.2-kartoza-build20230226-r9-allow-change-context-root-and-fix-missing-stable-plugins-and-avoid-chown-datadir`.

The following vulnerabilities where detected: 

- Critical:
  - https://app.snyk.io/vuln/SNYK-JAVA-COMH2DATABASE-2348247
  - https://app.snyk.io/vuln/SNYK-JAVA-ITGEOSOLUTIONSJAIEXTJIFFLE-2650766
  - https://app.snyk.io/vuln/SNYK-JAVA-XALAN-2953385
- High:
  - https://app.snyk.io/vuln/SNYK-JAVA-ORGSPRINGFRAMEWORKSECURITY-2833359
  - https://app.snyk.io/vuln/SNYK-JAVA-COMH2DATABASE-31685
  - https://app.snyk.io/vuln/SNYK-JAVA-COMH2DATABASE-2331071
  - https://app.snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-2331703
  - https://app.snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-3167772
  - https://app.snyk.io/vuln/SNYK-JAVA-ORGCODEHAUSJETTISON-3367610
  - https://app.snyk.io/vuln/SNYK-JAVA-ORGJYAML-550161
- Medium and below: ignored or duplicate of others above

For the most of them, updating latest packages (minor or patch revision) seems sufficient to fix the issues.

For `H2` database related issues, the fix requires a major change (`1.1.119` ->`2.1.210`). Maybe this can work out of the box. 

Otherwise, H2 seems to be used for Disk Quota, which can be switched to PostgreSQL, which we use for GeoServer.
https://github.com/kartoza/docker-geoserver#enable-disk-quota-storage-in-postgresql-backend

Therefore, the following plugin would not be required and could be removed:
https://github.com/kartoza/docker-geoserver/blob/a433c2d16729a52dbd82ebfd52db67ac93a6579b/build_data/stable_plugins.txt#L16

## References

- relates to #307

## Concerned Organizations

All

**Comment by tlvu** [comment]
Oh dang, how did you trigger this security scan? Can we scan a few more?  Like the Thredds image and our Jupyter image?

However, this issue should have been opened on Kartoza side since the custom image I build is simply the cache of Kartoza image with a few minor fixes, here's the Dockerfile: https://github.com/tlvu/docker-geoserver/blob/2e5dbb99effa75abe818cdf551532f2b1de7739c/Dockerfile.custom

**Comment by tlvu** [comment]
> Therefore, the following plugin would not be required and could be removed:
> https://github.com/kartoza/docker-geoserver/blob/a433c2d16729a52dbd82ebfd52db67ac93a6579b/build_data/stable_plugins.txt#L16

Oh they simply bundle all the known plugins in the docker image.  It does not means the plugin is enabled.

I have a feeling all these security issues are due to the various plugins, not Geoserver itself.

**Comment by tlvu** [comment]
> I found that there still is this Dockerfile in birdhouse-deploy: https://github.com/bird-house/birdhouse-deploy/blob/master/birdhouse/config/geoserver/Dockerfile
> I think it is deprecated and should be removed.

Exact.  This was before 2.19, so now that Dockerfile and all related files can be deleted.

**Comment by tlvu** [comment]
Are you able to trigger a scan on `pavics/geoserver:2.23.0-kartoza-build20230405` to see if the new tomcat base is there.  Else we'll upgrade for nothing.

**Comment by fmigneault** [comment]
> We already cached 2.23.0 so if you want to try, you can simply set `GEOSERVER_TAGGED=2.23.0-kartoza-build20230405`.

If it's working on Ouranos' side, I propose we directly update the `default.env` with it. You are most probably testing it more in depth than we do. 

I'll add it to the scan, and report back anything that comes up.

**Comment by fmigneault** [comment]
I've also started the scan directly in https://hub.docker.com/layers/pavics/geoserver/2.23.0-kartoza-build20230405/images/sha256-98eee4fc9c46fca45c9f56961f94366dacb65f65fa11d377ce337ba1a31683a1?context=explore, since DockerHub offers the same Snyk analysis out of the box.

**Comment by tlvu** [comment]
I think this is the result of your scan, so is the base image updated?
![Screenshot 2023-04-28 at 10-35-07 Image Layer Details - pavics_geoserver 2 23 0-kartoza-build20230405 Docker Hub](https://user-images.githubusercontent.com/11966697/235177133-cd7a1ae7-93de-4570-8214-f08a8c2e1100.png)

**Comment by fmigneault** [comment]
@tlvu 
Yes, that's the result of the scan.
It seems some updates were applied, but there are critical items that are still there. 
Notably, the `log4j/log4j 1.2.17` vulnerability is back in there.

**Comment by fmigneault** [comment]
Resolved a long time ago.
